airflow/api_fastapi/execution_api/datamodels/taskinstance.py

```diff
@@ -203,7 +203,7 @@ class TaskInstance(StrictBaseModel):
     dag_id: str
     run_id: str
     try_number: int
-    map_index: int = -1
+    map_index: int | None = None
```

Comment:
When can this be None? A TI should always have a map_index.

Reply:
Maybe the idea here is to have None instead of using -1 as a default?